### PR TITLE
Always use texture layer when displaying an Android view

### PR DESCRIPTION
**Pull request by blasten**
Ensures that Android views are always rendered using a texture layer.

Related issue: https://github.com/flutter/flutter/issues/96679

### CODE REVIEW

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 1]
```diff
@@ -74,7 +74,7 @@ Set<Type> _factoriesTypeSet<T>(Set<Factory<T>> factories) {
 ///
 ///  * [AndroidView] which is a widget that is used to show an Android view.
 ///  * [PlatformViewsService] which is a service for controlling platform views.
-class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
+class RenderAndroidView extends PlatformViewRenderBox {
```
Comment (blasten) [inline packages/flutter/lib/src/rendering/platform_view.dart:77]: Making `RenderAndroidView` extend `PlatformViewRenderBox` allows `AndroidViewSurface` to continue to extend `PlatformViewSurface` due to  the `updateRenderObject` override.

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 2]
```diff
@@ -101,23 +101,20 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
...
-    if (_viewController == viewController)
+    if (viewController == controller)
       return;
-    _viewController.removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
-    _viewController = viewController;
+
+    (controller as AndroidViewController).removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
+    controller = viewController;
+    viewController.addOnPlatformViewCreatedListener(_onPlatformViewCreated);
+    viewController.pointTransformer = (Offset offset) => globalToLocal(offset);
```
Comment (blasten) [inline packages/flutter/lib/src/rendering/platform_view.dart:116]: This seems to be bug currently. `pointTransformer` is set in the controller during initialization, but it's not set again if the controller is changed.

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 3]
```diff
@@ -101,23 +101,20 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
...
-    assert(_viewController != null);
-    assert(viewController != null);
-    if (_viewController == viewController)
+    if (viewController == controller)
       return;
-    _viewController.removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
-    _viewController = viewController;
+
+    (controller as AndroidViewController).removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
+    controller = viewController;
```
Comment (blasten) [inline packages/flutter/lib/src/rendering/platform_view.dart:114]: this calls the setter in the parent class. I don't like this API, but I'm trying to make this PR a non-breaking change.

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 4]
```diff
@@ -101,23 +101,20 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
...
   set viewController(AndroidViewController viewController) {
-    assert(_viewController != null);
-    assert(viewController != null);
-    if (_viewController == viewController)
+    if (viewController == controller)
       return;
-    _viewController.removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
-    _viewController = viewController;
+
+    (controller as AndroidViewController).removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
```
Comment (cyanglaz) [inline packages/flutter/lib/src/rendering/platform_view.dart:113]: Is `(controller as AndroidViewController)` just the getter `viewController`? (Not the parameter)

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 5]
```diff
@@ -101,23 +101,20 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
...
 
   /// The Android view controller for the Android view associated with this render object.
-  AndroidViewController get viewController => _viewController;
-  AndroidViewController _viewController;
+  AndroidViewController get viewController => controller as AndroidViewController;
+
   /// Sets a new Android view controller.
   ///
   /// `viewController` must not be null.
   set viewController(AndroidViewController viewController) {
```
Comment (cyanglaz) [inline packages/flutter/lib/src/rendering/platform_view.dart:109]: Prob better name this paramter so it doesn't conflict with the getter `viewController`

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 6]
```diff
@@ -101,23 +101,20 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
...
-    assert(viewController != null);
-    if (_viewController == viewController)
+    if (viewController == controller)
       return;
-    _viewController.removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
-    _viewController = viewController;
+
+    (controller as AndroidViewController).removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
+    controller = viewController;
+    viewController.addOnPlatformViewCreatedListener(_onPlatformViewCreated);
```
Comment (cyanglaz) [inline packages/flutter/lib/src/rendering/platform_view.dart:115]: It looks a little confusing changing the property after sets the ivar. How about:
```
this. viewController.removeOnPlatformViewCreatedListener(_onPlatformViewCreated);
controller = viewController
controller.addOnPlatformViewCreatedListener(_onPlatformViewCreated);
controller.pointTransformer = (Offset offset) => globalToLocal(offset);
...
```

File: packages/flutter/lib/src/services/platform_views.dart [hunk 7]
```diff
@@ -1093,17 +1056,21 @@ class TextureAndroidViewController extends AndroidViewController {
...
-  Future<void> create() => super.create();
+  Future<void> create() async {
+    if (_initialSize != null)
+      return super.create();
+  }
 
   @override
   Future<void> _sendCreateMessage() async {
-    assert(!_initialSize.isEmpty, 'trying to create $TextureAndroidViewController without setting a valid size.');
+    assert(_initialSize != null, 'trying to create $TextureAndroidViewController without setting an initial size.');
```
Comment (cyanglaz) [inline packages/flutter/lib/src/services/platform_views.dart:1066]: can we test these 2 asserts?

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 4 above)
Comment (cyanglaz) [inline packages/flutter/lib/src/rendering/platform_view.dart:113]: can this be tested? e.g. the listener is removed from the old controller. and added to the new controller

Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart]: I cleaned this up. I removed `viewController` and overrode `controller`.

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 5 above)
Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart]: I removed `viewController` and overrode `controller` instead.

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 6 above)
Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart]: done

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 4 above)
Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart]: Done in a7eeb3a

**Review by cyanglaz — APPROVED** [review]
LGTM

**Pull request merged**